packages/core/strapi/src/commands/actions/cloud/deploy-project/action.ts

```diff
@@ -0,0 +1,133 @@
+
+  if (!fs.existsSync(packageJsonPath)) {
+    logger.error(
+      "This project seems to be missing a 'package.json' file. It's required for project configuration."
+    );
+    return null;
+  }
+
+  try {
+    const packageJsonBuffer = fs.readFileSync(packageJsonPath);
+    const packageJson = JSON.parse(packageJsonBuffer.toString());
+
+    if (!packageJson.strapi || !packageJson.strapi.uuid) {
+      logger.error(
+        'This project appears to be missing a Strapi configuration. Please check your project setup and try again.'
+      );
+      return null;
+    }
+    return packageJson;
+  } catch (error: Error | unknown) {
+    if (error instanceof Error && error.message.includes('Unexpected token')) {
+      logger.error(
+        'The package.json file seems to have invalid formatting. Please check for syntax errors and try again.'
+      );
+    } else {
+      logger.error('Unable to proceed: The package.json file does not contain valid JSON data.');
+    }
+    logger.debug(error);
+  }
+  return null;
+}
+
+async function upload(ctx: CLIContext, project: ProjectInfos, token: string) {
+  const cloudApi = cloudApiFactory(token);
+  // * Upload project
+  try {
+    const storagePath = getStoragePath();
+    const projectFolder = path.resolve(process.cwd());
+    const packageJson = getPackageJson(ctx, projectFolder);
+
+    if (!packageJson) {
+      return;
+    }
+
+    ctx.logger.log('📦 Compressing project...');
+    const compressedFilename = `${packageJson.strapi.uuid}.tar.gz`;
+    try {
+      await compressFilesToTar(storagePath, projectFolder, compressedFilename);
+      ctx.logger.log('📦 Project compressed successfully!');
+    } catch (error: unknown) {
+      ctx.logger.error(
+        '⚠️ Project compression failed. Try again later or check for large/incompatible files.'
+      );
+      ctx.logger.debug(error);
+    }
+
+    const tarFilePath = path.resolve(storagePath, compressedFilename);
+    const fileStats = fs.statSync(tarFilePath);
+
+    if (fileStats.size > FILE_SIZE_LIMIT) {
+      return ctx.logger.log(
+        'Unable to proceed: Your project is too big to be transferred, please use a git repo instead.'
+      );
+    }
+
+    ctx.logger.info('🚀 Uploading project...');
+    const progressBar = ctx.logger.progressBar(100, 'Upload Progress');
+
+    try {
+      await cloudApi.deploy(
+        { filePath: tarFilePath, project },
+        {
+          onUploadProgress(progressEvent) {
+            const total = progressEvent.total || fileStats.size;
+            const percentage = Math.round((progressEvent.loaded * 100) / total);
+            progressBar.update(percentage);
+          },
+        }
+      );
+      progressBar.update(100);
+      ctx.logger.success('✨ Upload finished!');
+    } catch (error: any) {
+      ctx.logger.error('An error occurred while deploying the project. Please try again later.');
+      ctx.logger.debug(error);
+    } finally {
+      progressBar.stop();
+    }
```

Comment:
We miss a part here to display a link to the project page of the user on Strapi Cloud.
Also, we may add the logs from the build-engine streamlined directly into the CLI.
To be decided/implemented by next Wednesday (April 17th).